build-info-api/src/main/java/org/jfrog/build/api/builder/DependencyBuilder.java

```diff
@@ -161,27 +164,29 @@ public DependencyBuilder remotePath(String remotePath) {
     }
 
     /**
-     * Sets an ID list of other dependencies required by this one
+     * Sets path-to-root of lists of dependencies that directly depend on this dependency.
      *
-     * @param requiredBy Required dependency IDs list
+     * @param requiredBy dependency path-to-root list
      * @return Builder instance
      */
-    public DependencyBuilder requiredBy(List<String> requiredBy) {
+    public DependencyBuilder requiredBy(String[][] requiredBy) {
         this.requiredBy = requiredBy;
         return this;
     }
 
     /**
      * Adds an ID of another dependency required by this one to the required dependencies list
      *
-     * @param requiredBy Required dependency ID
+     * @param pathToModuleRoot - path from parent dependency to the module ID
      * @return Builder instance
      */
-    public DependencyBuilder addRequiredBy(String requiredBy) {
-        if (this.requiredBy == null) {
-            this.requiredBy = new ArrayList<>();
+    public DependencyBuilder addRequiredBy(String[] pathToModuleRoot) {
+        if (requiredBy == null) {
+            requiredBy = new String[][]{pathToModuleRoot};
+        } else {
+            requiredBy = Arrays.copyOf(requiredBy, requiredBy.length + 1);
+            requiredBy[requiredBy.length - 1] = Arrays.copyOf(pathToModuleRoot, pathToModuleRoot.length);
```

Comment:
We should use this in `build-info-api/src/main/java/org/jfrog/build/api/Dependency.java` to add a new path as well as here

Reply:
Done